lib/LedgerSMB/Admin/Command/destroy.pm

```diff
@@ -0,0 +1,110 @@
+
+package LedgerSMB::Admin::Command::destroy;
+
+=head1 NAME
+
+LedgerSMB::Admin::Command::destroy - ledgersmb-admin 'destroy' command
+
+=cut
+
+use strict;
+use warnings;
+
+use LedgerSMB::Admin::Command;
+use LedgerSMB::Database;
+
+use Moose;
+extends 'LedgerSMB::Admin::Command';
+use namespace::autoclean;
+
+sub run {
+    my ($self, $dbname, $newname) = @_;
+    my $logger = $self->logger;
+    my $existing_db = $self->connect_data_from_arg($dbname);
+    my $connect_data = {
+        $self->config->get('connect_data')->%*,
+        $existing_db->%*,
+    };
+    my $db = LedgerSMB::Database->new(
+        connect_data => $connect_data,
+        );
+    my $dbh = $db->connect;
+    $dbh->do(q{SELECT setting__set('role_prefix',
+                             coalesce((setting_get('role_prefix')).value, ?))},
+                 {},
+                 "lsmb_$existing_db->{dbname}__")
+        or die $dbh->errstr;
+    $dbh->commit;
+    $dbh->disconnect;
```

Comment:
Why are you *setting* the value of `role_prefix` if you're about to delete the database?

Reply:
Indeed. That had to be a get only.